mailet/api/src/main/java/org/apache/mailet/Serializer.java

```diff
@@ -0,0 +1,473 @@
+    JsonNode serialize(T object);
+
+    Optional<T> deserialize(JsonNode json);
+
+    String getName();
+
+    class Registry {
+
+        private static ImmutableMap<String, Serializer<?>> serializers;
+
+        static {
+            serializers = Stream
+                .<Serializer<?>>of(
+                    BOOLEAN_SERIALIZER,
+                    STRING_SERIALIZER,
+                    INT_SERIALIZER,
+                    LONG_SERIALIZER,
+                    FLOAT_SERIALIZER,
+                    DOUBLE_SERIALIZER,
+                    new Serializer.ArbitrarySerializableSerializer(),
+                    URL_SERIALIZER,
+                    new CollectionSerializer<>(),
+                    new MapSerializer<>(),
+                    new FSTSerializer(),
+                    new OptionalSerializer<>())
+                .collect(ImmutableMap.toImmutableMap(Serializer::getName, Function.identity()));
+        }
+
+        static Optional<Serializer<?>> find(String name) {
+            return Optional.ofNullable(serializers.get(name));
+        }
+    }
+
+    class BooleanSerializer implements Serializer<Boolean> {
+        @Override
+        public JsonNode serialize(Boolean object) {
+            return BooleanNode.valueOf(object);
+        }
+
+        @Override
+        public Optional<Boolean> deserialize(JsonNode json) {
+            if (json instanceof BooleanNode) {
+                return Optional.of(json.asBoolean());
+            } else {
+                return Optional.empty();
+            }
+        }
+
+        @Override
+        public String getName() {
+            return "BooleanSerializer";
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            return this.getClass() == other.getClass();
+        }
+    }
+
+    Serializer<Boolean> BOOLEAN_SERIALIZER = new BooleanSerializer();
+
+    class StringSerializer implements Serializer<String> {
+        @Override
+        public JsonNode serialize(String object) {
+            return TextNode.valueOf(object);
+        }
+
+        @Override
+        public Optional<String> deserialize(JsonNode json) {
+            if (json instanceof TextNode) {
+                return Optional.of(json.asText());
+            } else {
+                return Optional.empty();
+            }
+        }
+
+        @Override
+        public String getName() {
+            return "StringSerializer";
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            return this.getClass() == other.getClass();
+        }
+    }
+
+    Serializer<String> STRING_SERIALIZER = new StringSerializer();
+
+    class IntSerializer implements Serializer<Integer> {
+        @Override
+        public JsonNode serialize(Integer object) {
+            return IntNode.valueOf(object);
+        }
+
+        @Override
+        public Optional<Integer> deserialize(JsonNode json) {
+            if (json instanceof NumericNode) {
```

Comment:
`NumericNode` seems not is a tight condition, how about `IntNode`?
And similar to other Serializers